avoid calling `conductor`

ThomasBreuer opened the pull request:
Instead, call `Hecke.force_coerce_cyclo(Kg, data(x), Val(false))`, which finds out whether the conversion works.

ThomasBreuer committed:
avoid calling `conductor`

Instead, call `Hecke.force_coerce_cyclo(Kg, data(x), Val(false))`,
which finds out whether the conversion works.

thofma commented on src/Rings/AbelianClosure.jl:
```diff
@@ -933,25 +933,22 @@ end
-      error("element has no preimage")
-    end
+    res = Hecke.force_coerce_cyclo(C1, data(x), Val(false))
+    !check && res === nothing && error("element has no preimage")
```
Hm, if `check == true` and `res === nothing`, will this return nothing? Sorry, I am not good with logic.

fingolfin replied:
This code is rather confusing to me as well. But so is the old code: it indeed also returns `nothing` when `check` is `true` and `conductor(x) != 1` ?!

fingolfin replied:
I guess what confuses me is that `check` apparently has a different meaning here than it has in e.g. our constructors. Since I don't know what the meaning of `check` is supposed to be here, I can't comment further.

Is it documented somewhere?

ThomasBreuer replied:
@fingolfin See my comment below: The function in question is internal and not documented (as well as the functions called by it). I think we could change the meaning of `check` to that in other situations.

thofma commented:
might be better if @fieker has a look

ThomasBreuer commented:
Yes, the function `_embedding` returns `nothing` if the second argument suppresses the error, like `Hecke.force_coerce_cyclo` does.
(I admit that the meaning of `check` may be confusing; taking that `_embedding` is undocumented and used only in my code, we could change this.)

thofma commented:
Sorry, I missed that this is an internal function. Then forget my question. But @fieker should check.

ThomasBreuer committed:
try to reduce some confusion

changed a `check = false` keyword argument to `throw_error = true`
in an internal function, `check` was a misleading name;
the keyword argument is used only by a `has_preimage_with_preimage`
method

ThomasBreuer commented:
@fingolfin @fieker I have changed the name of the keyword argument from `check` to `throw_error`. Since this is (hopefully) used only by a `has_preimage_with_preimage` method, this change should not be visible outside.
And it does not affect the proposed improvement.

ThomasBreuer commented:
One failure due to a segmentation fault, restarted this test.

fingolfin reviewed (approved):
Looks good to me

fingolfin commented:
@fieker please merge if you are OK with this versio
